packages/ui/src/IconButton.tsx

```diff
@@ -1,183 +1,155 @@
-import React, {forwardRef, useState} from "react";
-import {Platform, Pressable, View, ViewStyle} from "react-native";
+import FontAwesome6 from "@expo/vector-icons/FontAwesome6";
+import debounce from "lodash/debounce";
+import React, {FC, useContext, useState} from "react";
+import {ActivityIndicator, Pressable, View} from "react-native";
 
-import {Box} from "./Box";
-import {IconButtonProps, IconColor, iconSizeToNumber, SurfaceColor} from "./Common";
-import {Icon} from "./Icon";
+import {IconButtonProps} from "./Common";
+import {isMobileDevice} from "./MediaQuery";
 import {Modal} from "./Modal";
 import {Text} from "./Text";
+import {ThemeContext} from "./Theme";
 import {Tooltip} from "./Tooltip";
 import {Unifier} from "./Unifier";
+import {isNative} from "./Utilities";
 
-// eslint-disable-next-line react/display-name
-export const IconButton = forwardRef(
-  (
-    {
-      prefix,
-      icon,
-      onClick,
-      size,
-      type = "primary",
-      withConfirmation = false,
-      confirmationText = "Are you sure you want to continue?",
-      confirmationHeading = "Confirm",
-      tooltip,
-      indicator,
-      indicatorNumber,
-      indicatorStyle = {position: "bottomRight", color: "primary"},
-      testID,
-    }: IconButtonProps,
-    ref
-  ) => {
-    const [showConfirmation, setShowConfirmation] = useState(false);
+const ConfirmationModal: FC<{
```

Comment:
I think the preferred way of typing components is `({prop1}: ComponentProps): ReturnType`

If we have a good reason to change the standard, I'm definitely open but I think it'd be best to keep it consistent.

Reply:
I meant to update this. Pushing up a change i had similar to a different component I pushed out. Lemme know whatcha think

Reply:
I can type ComponentProps but it will remain local to this file since the confirmation modal has already has a circular call and i'm trying to avoid app crashing on render

We really should use FC instead or react.element since we are returning components and ReactElememt  the return type instead of the component itself.
A GPT explanation:
`type FC<P = {}> = FunctionComponent<P>;`
Declaring as a FC<Props> provides type safety for the component's props. FC (Functional Component) is a type provided by React that includes the type definition for props, children, and some default properties. It ensures that the component will receive props of type Props.

Component Definition:
Using FC defines Button as a functional component. A functional component is a function that takes props as an argument and returns a ReactElement. This is exactly what the Button component is doing. The component logic is defined in a function, and it returns different ReactElement structures based on the conditions within.

A ReactElement is the result of invoking a React component. It is the return type of a component, not the component itself. The Button function returns a ReactElement, but it is not itself a ReactElement. By defining Button as a FC, we ensure that it adheres to the structure and behavior of a functional component.

FC automatically includes default properties and children, adding a layer of type safety and consistency. This is beneficial for maintaining standard practices and ensuring the component can handle children and other default React properties without additional type declarations.

Reply:
I think Hannah's point makes sense but I do think that if we do this, we should change it everywhere else as well and have style consistency throughout the codebase. So just want to make sure we go back and change for components we already updated and make sure we stick to this moving forward.

Reply:
Thanks for sharing the explanation. I was researching this a little bit and came across this: https://github.com/facebook/create-react-app/pull/8177.

It's essentially pulling the FC standard out of the Typescript boilerplate for a react project because of some smaller issues that it can cause. Granted, I don't know that those issues would apply here directly but I'm just looking for consistency. I'm good with keeping it in here if the circular dependencies are causing a headache but I think we should try to use the standard we're currently using throughout our codebase if there isn't a specific reason to change it.

Also happy to bring this to #dev if we want some more opinions.